services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/table/TableQueryManagerImplTest.java

```diff
@@ -836,13 +847,52 @@ public void testQueryAsStreamAfterAuthorizationWithSearchDisabled() throws Excep
 		
 		String message = assertThrows(IllegalArgumentException.class, () -> {			
 			// call under test
-			manager.queryAsStreamAfterAuthorization(mockProgressCallbackVoid, query, rowHandler, queryOptions);
+			manager.queryAsStreamAfterAuthorization(user, mockProgressCallbackVoid, query, rowHandler, queryOptions);
 		}).getMessage();
 		
 		assertEquals("Invalid use of TEXT_MATCHES. Full text search is not enabled on table syn123.", message);
 
 	}
 	
+	@Test
+	public void testQueryAsStreamAfterAuthorizationWithActionsRequired() throws Exception {
+		when(mockTableConnectionFactory.getConnection(idAndVersion)).thenReturn(mockTableIndexDAO);
+		when(mockTableIndexDAO.getConnection()).thenReturn(mockJdbcTemplate);
+		when(mockSchemaProvider.getTableSchema(any())).thenReturn(models);
+		setupQueryCallback();
+		when(mockTableIndexDAO.executeInWriteTransaction(any())).thenAnswer(new Answer<Object>() {
+			@Override
+			public Object answer(InvocationOnMock invocation) throws Throwable {
+				return invocation.getArgument(0, TransactionCallback.class).doInTransaction(null);
+			}
+		});
+		
+		List<ActionRequiredCount> actions = List.of(new ActionRequiredCount().setCount(3L).setAction(new MeetAccessRequirement().setAccessRequirementId(345L)));

```

Comment:
what could we do to test this with a mockActionsRequiredDao, instead of testing the JDBCTemplate?